generator/src/main/java/com/scottlogic/deg/generator/generation/FieldSpecValueGeneratorFactory.java

```diff
@@ -0,0 +1,41 @@
+package com.scottlogic.deg.generator.generation;
+
+import com.scottlogic.deg.generator.Field;
+import com.scottlogic.deg.generator.fieldspecs.FieldSpec;
+import com.scottlogic.deg.generator.generation.databags.DataBag;
+import com.scottlogic.deg.generator.generation.databags.DataBagSource;
+
+import java.util.stream.Stream;
+
+public class FieldSpecValueGeneratorFactory {
+    private final FieldValueSourceEvaluator evaluator;
+
+    public FieldSpecValueGeneratorFactory(FieldValueSourceEvaluator evaluator) {
+        this.evaluator = evaluator;
+    }
+
+    public DataBagSource getFieldSpecValueGenerator(Field field, FieldSpec fieldSpec){
```

Comment:
This class is an adaptor, can it be named as such and moved to its own file

Reply:
Renamed to `FieldSpecValueGeneratorAdapter` given the description:

> Adapter or wrapper: Converts one interface to another so that it matches what the client is expecting
https://en.wikipedia.org/wiki/Adapter_pattern